models/events.js

```diff
@@ -237,11 +237,46 @@ const createEventCode = async (eventCodeData) => {
   }
 };
 
+const getEventById = async ({ id }) => {
+  try {
+    const eventRef = eventModel.doc(id);
+    const eventSnapshot = await eventRef.get();
+    const eventSnapshotData = eventSnapshot.data();
+    return eventSnapshotData;
+  } catch (error) {
+    logger.error("Error in getting event by id from the database.", error);
+    throw error;
+  }
+};
+
+const getEventCodes = async ({ id }) => {
```

Comment:
eventCodesArrayPromises array is not being checked for empty before it is passed to the Promise.all() function. This means that if the array is empty, the Promise.all() function will still wait for all the promises to resolve, even though there are no promises.

Reply:
resolved this one